packages/gestalt/src/SideNavigation/NavigationContent.js

```diff
@@ -0,0 +1,149 @@
+// @flow strict
+import { type Node as ReactNode, useEffect, useRef, useState } from 'react';
+import classnames from 'classnames';
+import Collapser from './Collapser';
+import {
+  countItemsWithIcon,
+  groupIconlessChildren,
+  validateChildren,
+} from './navigationChildrenUtils';
+import borderStyles from '../Borders.css';
+import Box from '../Box';
+import boxStyles from '../Box.css';
+import { useSideNavigation } from '../contexts/SideNavigationProvider';
+import Divider from '../Divider';
+import Flex from '../Flex';
+import layoutStyles from '../Layout.css';
+import { type Props as SideNavigationProps } from '../SideNavigation';
+import styles from '../SideNavigation.css';
+import { flattenChildrenWithKeys } from '../utils/flattenChildren';
+import { FixedZIndex } from '../zIndex';
+
+type Props = { ...SideNavigationProps };
+type Timeout = ReturnType<typeof setTimeout>;
+
+export default function NavigationContent({
+  accessibilityLabel,
+  children,
+  footer,
+  header,
+  showBorder,
+}: Props): ReactNode {
+  const navigationChildren = flattenChildrenWithKeys(children);
+
+  validateChildren({ children: navigationChildren, filterLevel: 'main' });
+
+  const {
+    collapsible,
+    collapsed: sideNavigationCollapsed,
+    onCollapse,
+    overlayPreview,
+    setOverlayPreview,
+    transitioning,
+  } = useSideNavigation();
+
+  const scrollContainer = useRef<HTMLDivElement | null>(null);
```

Comment:
Follow up exploration: can we make consistent shadows in scrolling? See Modal.  Are we considering here resizing not just scrolling?

![Screenshot by Dropbox Capture](https://github.com/pinterest/gestalt/assets/10593890/19692361-e8ef-4222-a119-a604f1ed45d2)

Reply:
good question. I will make a follow up PR for this.

Reply:
@AlbertCarreras No need to follow up. Because SideNavigation is not responsive. It has fixed 280px width when it's collapsible. So resizing doesn't cause content to shift.